### PR TITLE
Linter messages are considered an error

### LINKED ISSUE #107: [conan center] fail on linter warnings
disclaimer: not sure if it belongs here at all
currently, some cci recipes have linter warnings, e.g. `libpng`:
```
Linter warnings
    WARN: Linter. Line 6: Unused ConanException imported from conans.errors
```
it would be nice if hook rejects such recipes

### PR DESCRIPTION
**Pull request by uilianries**
If the message _Linter warnings_ is present in the output stream, so this hook will raise an error.

We could empathize the linter messages, but it will consumes some new lines in middle of the output and it will be duplicated.

closes #107

Wiki:

#### **<a name="KB-H026">#KB-H026</a>: "LINTER WARNINGS"**

Conan is equipped with [Python linter](https://docs.conan.io/en/latest/integrations/linting/recipe.html) which can find prone errors in your recipe.
All warnings reported by the linter must be fixed to accept your recipe in CCI.

### CODE REVIEW

File: tests/test_hooks/conan-center/test_conan-center.py [hunk 1]
```diff
@@ -399,3 +402,17 @@ def test_cmake_minimum_verstion(self):
...
+        from conans import ConanFile, tools, CMake
+        import os
+        import platform
+
+        class AConan(ConanFile):
+            pass
+        """)
+        tools.save('conanfile.py', content=conanfile)
+        output = self.conan(['create', '.', 'name/version@jgsogo/test'])
+        self.assertIn("ERROR: [LINTER WARNINGS (KB-H026)] This recipe is not clear according " \
```
Comment (danimtb) [inline tests/test_hooks/conan-center/test_conan-center.py]: Additionally, check the linter warning lines in the output if possible

File: hooks/conan-center.py [hunk 2]
```diff
@@ -299,6 +300,13 @@ def test(out):
                 out.error("Use 'tools.get(**self.conan_data[\"sources\"][\"XXXXX\"])' "
                           "in the source() method to get the sources.")
 
+    @run_test("KB-H026", output)
+    def test(out):
+        # INFO: _HooksOutputErrorCollector->ScopedOutput->StringIO to str
+        out_stream = str(output._output._stream.getvalue())
+        if "Linter warnings" in out_stream:
+            out.error("This recipe is not clear according Conan Linter")
```
[the remark this answers is not in the document]
Comment (uilianries) [reply hooks/conan-center.py]: nice! much better!

File: tests/test_hooks/conan-center/test_conan-center.py [hunk 3]
```diff
@@ -399,3 +402,19 @@ def test_cmake_minimum_verstion(self):
...
+        import platform
+
+        class AConan(ConanFile):
+            pass
+        """)
+        tools.save('conanfile.py', content=conanfile)
+        output = self.conan(['create', '.', 'name/version@jgsogo/test'])
+        self.assertIn("ERROR: [LINTER WARNINGS (KB-H026)] Linter warnings detected." \
+                      " Check the warnings in the output and fix them in the recipe", output)
+        self.assertIn("Linter warnings", output)
```
Comment (danimtb) [inline tests/test_hooks/conan-center/test_conan-center.py]: not needed?

Comment (uilianries) [reply tests/test_hooks/conan-center/test_conan-center.py]: do you mean because it is replicated with the hook?

Comment (danimtb) [reply tests/test_hooks/conan-center/test_conan-center.py]: I think it is a redundant check but it doesn`t harm. Don't worry

**Comment by danimtb** [comment]
I am having a look at this hook and we have already a built-in option in Conan for this: `pylint_werr = True`.

We should revert this hook

**Comment by danimtb** [comment]
Additionally, this is causing some trouble in the CI, so I will revert it and consider its inclusion in the future if the `pylint_werr` is inconvenient

**Comment by uilianries** [comment]
> `pylint_werr`

It sounds better, if you can change it directly on pylint, do it.

**Pull request merged**